website/docs/r/log_search.html.markdown

```diff
@@ -85,6 +100,16 @@ The following arguments are supported:
 - `run_by_receipt_time` - (Optional) This has the value `true` if the search is to be run by receipt time and
     `false` if it is to be run by message time. Default value is `false`.
 
+### Schema for `query_parameter`
+- `name` - (Required) The name of the parameter.
+- `description` - (Optional) A description of the parameter.
+- `dataType` - (Required) The data type of the parameter. Supported values are:
+  1. `NUMBER`
+  2. `STRING`
+  3. `ANY`
+  4. `KEYWORD`
+- `value` - (Required) A value for the parameter. Should be compatible with the type set in dataType field.
```

Comment:
I don't see how this lines up with the [documentation](https://help.sumologic.com/docs/search/get-started-with-search/build-search/search-templates/). 

Here's what I see when I create a query parameter in the UI.
![image](https://github.com/SumoLogic/terraform-provider-sumologic/assets/76980899/3f8a88c6-cf3a-47ad-a57f-645594c8b0b6)

When I select "Manage Parameter Settings" I get this:
![image](https://github.com/SumoLogic/terraform-provider-sumologic/assets/76980899/e95ae7fa-ad15-43ae-8c4c-d780ceb8576d)

So at this point does `value` represent the "Default Value" you see in the above screenshot? If not, then I assume it address the "Set Values for Parameter" settings. Here is what that looks like:
![image](https://github.com/SumoLogic/terraform-provider-sumologic/assets/76980899/8d7d0cf0-8f35-4d79-acf6-f97f0dcf17cb)
Multiple options for format:
![image](https://github.com/SumoLogic/terraform-provider-sumologic/assets/76980899/e4afcad6-418e-401c-b1c9-c0a62e211266)
Let's say I select "Lookup". Now how would I go about filling out these options using this Terraform provider?
![image](https://github.com/SumoLogic/terraform-provider-sumologic/assets/76980899/e1852a06-bd26-4526-9a3c-ed546c727544)

Reply:
I did the local testing for you and have results...

This `value` is in fact the "Default Value" you see in the UI. The provider should be updated to rename this field correctly to reflect that, otherwise it's confusing. I recommend `default_value`.

Reply:
The provider code and api expect the parameter name to be value. Have added the details in the description of the parameter.

Reply:
The `Set Values for Parameter ` part is not supported via the log search api so it wont be supported through terraform. The `value` is used as default value. Added this detail in the description for `value` parameter.

Reply:
I believe that means that we only support the default format of `None`

![image](https://github.com/SumoLogic/terraform-provider-sumologic/assets/76980899/c38bae00-b2d1-443d-8368-75e609482b25)

I think it's worth adding a note in this resource that setting autocomplete values for a query_parameter is not supported at this time.

Reply:
It would be good to test if it can be set in the UI after the resource is created with TF, and that TF won't reset those values afterwards. That at least provides somewhat of a workaround and a partial way to define those types of parameters as code.

Reply:
Tested this with terraform. The log search was created with the default value as mentinoed in terraform. I was further able to add the dropdown list values for the parameter using UI and on running terraform plan again it does not show any changes after this.